python/pyspark/pandas/supported_api_gen.py

```diff
@@ -0,0 +1,356 @@
+            f"{ps_module_path}.{module_dot_func}.html>`__ for detail."
+        )
+        missing_str += additional_str
+    return missing_str
+
+
+def get_pd_modules(pd_module_group: Any) -> List[str]:
+    """
+    Returns sorted pandas memeber list from pandas module path.
+
+    Parameters
+    ----------
+    pd_module_group : Any
+        Specific path of importable pandas module.
+    """
+    return sorted([m[0] for m in getmembers(pd_module_group, isclass) if not m[0].startswith("_")])
+
+
+def update_all_supported_status(
+    all_supported_status: Dict[Tuple[str, str], Dict[str, SupportedStatus]],
+    pd_modules: List[str],
+    pd_module_group: Any,
+    ps_module_group: Any,
+) -> None:
+    """
+    Updates supported status across multiple module paths.
+
+    Parmeters
+    ---------
+    all_supported_status: Dict[Tuple[str, str], Dict[str, SupportedStatus]]
+        Data that stores the supported status across multiple module paths.
+    pd_modles: List[str]
+        Name list of pandas modules.
+    pd_module_group : Any
+        Specific path of importable pandas module.
+    ps_module_group: Any
+        Specific path of importable pyspark.pandas module.
+    """
+    pd_modules += [""]  # for General Function APIs
+    for module_name in pd_modules:
+        supported_status = create_supported_by_module(module_name, pd_module_group, ps_module_group)
+        if supported_status:
+            all_supported_status[(module_name, ps_module_group.__name__)] = supported_status
+
+
+def write_table(
+    module_name: str,
+    module_path: str,
+    supported_status: Dict[str, SupportedStatus],
+    w_fd: TextIO,
+) -> None:
+    """
+    Write table by using Sphinx list-table directive.
+    """
+    lines = []
+    lines.append("Supported ")
+    if module_name:
+        lines.append(module_name)
+    else:
+        lines.append("General Function")
+    lines.append(" APIs\n")
+    lines.append("-" * 100)
+    lines.append("\n")
+    lines.append(f".. currentmodule:: {module_path}")
+    if module_name:
+        lines.append(f".{module_name}\n")
+    else:
+        lines.append("\n")
+    lines.append("\n")
+    lines.append(".. list-table::\n")
+    lines.append("    :header-rows: 1\n")
+    lines.append("\n")
+    lines.append("    * - API\n")
+    lines.append("      - Implemented\n")
+    lines.append("      - Missing parameters\n")
+    for func_str, status in supported_status.items():
+        func_str = escape_func_str(func_str)
+        if status.implemented == Implemented.NOT_IMPLEMENTED.value:
+            lines.append(f"    * - {func_str}\n")
+        else:
+            lines.append(f"    * - :func:`{func_str}`\n")
+        lines.append(f"      - {status.implemented}\n")
+        lines.append("      - \n") if not status.missing else lines.append(
+            f"      - {status.missing}\n"
+        )
+    w_fd.writelines(lines)
+
+
+def escape_func_str(func_str: str) -> str:
```

Comment:
Looks like we need some special process in here for:
- Link to alias link, such as `Dataframe.sub/subtract`
- Link to parent method, such as `Dataframe.ewm`, `SeriesGroupBy` doc

Otherwise, the gen link has some problem, feel free to do it in a separate PR.

Reply:
@Yikun That's a good point. This function was added because pandas' "is_" function (https://pandas.pydata.org/docs/reference/api/pandas.Index.is_.html) prevented the 'rst' format from being created. All "is_" functions are not implemented in pyspark, so we do not have to worry about links for now, but it may become an issue in the future, so I will leave a comment and make it possible to work in other PRs.